### ISSUE TITLE
Can only delete and create an interface once - opx-nas service crashes and does not restart

### ISSUE THREAD

**Issue by dimbleby** [open-switch/opx-cps#53]
As discussed in #51.  Here's a session on a 2.2 VM:

```
opxUser@opx_vm:~$ sudo cps_set_oid.py -qua target -oper action dell-base-if-cmn/set-interface dell-base-if-cmn/set-interface/input/operation=2 if/interfaces/interface/name=e101-002-0
Success
Key: 1.19.1245192.
cps/object-group/return-code = 0
dell-base-if-cmn/set-interface/input/operation = 2
if/interfaces/interface/name = e101-002-0
opxUser@opx_vm:~$
opxUser@opx_vm:~$ sudo cps_set_oid.py -qua target -oper action dell-base-if-cmn/set-interface dell-base-if-cmn/set-interface/input/operation=1 if/interfaces/interface/name=e101-002-0 if/interfaces/interface/type=ianaift:ethernetCsmacd
Success
Key: 1.19.1245192.
dell-base-if-cmn/if/interfaces/interface/if-index = 37
if/interfaces/interface/name = e101-002-0
if/interfaces/interface/type = ianaift:ethernetCsmacd
cps/object-group/return-code = 0
dell-base-if-cmn/set-interface/input/operation = 1
dell-if/if/interfaces/interface/negotiation = 1
opxUser@opx_vm:~$
opxUser@opx_vm:~$ sudo cps_set_oid.py -qua target -oper action dell-base-if-cmn/set-interface dell-base-if-cmn/set-interface/input/operation=2 if/interfaces/interface/name=e101-002-0
opxUser@opx_vm:~$
```

Notice that the third command - the second attempt to delete the interface - does not succeed.

**Comment by atanu-mandal** [comment]
I am curious to understand the use case of this kind of operations. Can you explain little bit on where it can be applied to. Thanks.

**Comment by dimbleby** [comment]
This isn't something that I'd expect to be a common operation - it's just a failure that arose during our testing.

**Comment by dimbleby** [comment]
It seems that this sequence of commands causes the `opx-nas` service to crash:

```
opxUser@opx_vm:~$ sudo service opx-nas status
● opx-nas.service - Network abstraction service
   Loaded: loaded (/lib/systemd/system/opx-nas.service; enabled)
   Active: failed (Result: signal) since Sat 2018-02-03 10:16:12 UTC; 2s ago
  Process: 670 ExecStart=/usr/bin/opx_nas_daemon (code=killed, signal=SEGV)
  Process: 571 ExecStartPre=/usr/bin/dn_rules.sh (code=exited, status=0/SUCCESS)
 Main PID: 670 (code=killed, signal=SEGV)

Feb 03 10:15:58 opx_vm opx_nas_daemon[670]: [QOS:QOS], Allocate nas-sg-id 0x2102000600000025 for VP port_id 37, level 2, local_sg_index 6
Feb 03 10:15:58 opx_vm opx_nas_daemon[670]: [QOS:QOS], Allocate nas-sg-id 0x2102000700000025 for VP port_id 37, level 2, local_sg_index 7
Feb 03 10:15:58 opx_vm opx_nas_daemon[670]: [QOS:QOS], Allocate nas-sg-id 0x2102000800000025 for VP port_id 37, level 2, local_sg_index 8
Feb 03 10:15:58 opx_vm opx_nas_daemon[670]: [QOS:QOS], Allocate nas-sg-id 0x2102000900000025 for VP port_id 37, level 2, local_sg_index 9
Feb 03 10:16:12 opx_vm opx_nas_daemon[670]: [INTERFACE:INTF-C], Failed to get if_info
Feb 03 10:16:12 opx_vm opx_nas_daemon[670]: [INTERFACE:NAS-INT], Interface 37 returned error -2013167616
Feb 03 10:16:12 opx_vm systemd[1]: opx-nas.service: main process exited, code=killed, status=11/SEGV
Feb 03 10:16:12 opx_vm systemd[1]: Unit opx-nas.service entered failed state.
Feb 03 10:16:12 opx_vm systemd[1]: Triggering OnFailure= dependencies of opx-nas.service.
Feb 03 10:16:12 opx_vm systemd[1]: Failed to enqueue OnFailure= job: Invalid argument
```

I guess that does make it relatively serious - even though it would be an unusual thing to do, the result is quite severe.

**Comment by atanu-mandal** [comment]
We are root causing the behaviour.

**Comment by atanu-mandal** [comment]
It seems there is no such use case attached to this opearation for your application. It was observed during a stress testing, thanks for sharing the result though. We will follow up on this.

**Comment by atanu-mandal** [comment]
The expectation is to pass "front-panel-port" and "subport id" in the create request otherwise it will create interface with NPU_port which is virtual interface. So if below procedure is followed it would behave as expected.

cps_set_oid.py -qua target -oper action dell-base-if-cmn/set-interface dell-base-if-cmn/set-interface/input/operation=1 if/interfaces/interface/name=e101-002-0 if/interfaces/interface/type=ianaift:ethernetCsmacd  base-if-phy/hardware-port/front-panel-port=2 base-if-phy/hardware-port/subport-id=0

Given that this is not an usual test case/operation , we will close this bug.

**Comment by dimbleby** [comment]
Shouldn't the switch be resilient to mistakes?

That is: wouldn't it be better to:
* fix the bug
* arrange that the opx-nas service can restart after this and any similar failure?

Understood that this is in some sense user error (originating with you actually! see [this comment](https://github.com/open-switch/opx-cps/issues/51#issuecomment-362503249)) and therefore low priority - but still, I'm a little surprised that you are happy to close it out without making any fix...?

**Comment by jeff-yin** [comment]
Actually the issue is being fixed so that nas will not crash in this scenario, but given that it's an uncommon operation, we'll roll out the fix in a future release of OPX. I.e., the fix will be a part of a "stable" release rather than a "testing" release. 

We can leave the issue open for now for tracking purposes.

**Comment by atanu-mandal** [comment]
@dimbleby , definitely. It was my bad, given that the information was provided for proper execution of the operation (to unblock any effort)  I had closed this one. We will provide you more update when the crash is actually fixed.

**Comment by jeff-yin** [comment]
This issue should be fixed in the latest release of OPX. Closing for now. Reopen if you do see it again.
